src/Core/Models/Business/SecretsManagerSubscriptionUpdate.cs

```diff
@@ -88,7 +88,7 @@ public class SecretsManagerSubscriptionUpdate
         SmServiceAccountsExcludingBase = newTotalServiceAccounts - secretsManagerPlan.BaseServiceAccount.GetValueOrDefault();
         MaxAutoscaleSmServiceAccounts = maxAutoscaleServiceAccounts;
 
-        MaxAutoscaleSmSeatsChanged = maxAutoscaleSeats.HasValue && maxAutoscaleSeats != organization.MaxAutoscaleSmSeats.GetValueOrDefault();
-        MaxAutoscaleSmServiceAccountsChanged = maxAutoscaleServiceAccounts.HasValue && maxAutoscaleServiceAccounts != organization.MaxAutoscaleSmServiceAccounts.GetValueOrDefault();
+        MaxAutoscaleSmSeatsChanged = maxAutoscaleSeats != organization.MaxAutoscaleSmSeats;
+        MaxAutoscaleSmServiceAccountsChanged = maxAutoscaleServiceAccounts != organization.MaxAutoscaleSmServiceAccounts;
```

Comment:
These are all nullable, we need to use `GetValueOrDefault`. Per #3085:

```cs
           MaxAutoscaleSmSeatsChanged =
                MaxAutoscaleSeats.GetValueOrDefault() != organization.MaxAutoscaleSmSeats.GetValueOrDefault(),
            MaxAutoscaleSmServiceAccountsChanged =
                MaxAutoscaleServiceAccounts.GetValueOrDefault() != organization.MaxAutoscaleSmServiceAccounts.GetValueOrDefault()
```

Reply:
`.GetValueOrDefault()` doesn’t compare nulls, they default to zero, and we want to compare if `null == 0`